modules/tinymce/src/core/main/ts/keyboard/Nbsps.ts

```diff
@@ -54,6 +55,25 @@ const isAtLineBoundary = (root: SugarElement, pos: CaretPosition) => (
     isBeforeBr(root, pos)
 );
 
+const isCefBlock = (node?: Node) =>
+  Type.isNonNullable(node) && NodeType.isContentEditableFalse(node) && isBlockLike(node);
+
+const isBeforeCefBlock = (root: SugarElement, pos: CaretPosition) => {
+  if (pos.isAtEnd()) {
+    return isCefBlock(new DomTreeWalker(pos.container(), root.dom).next());
+  } else {
+    return false;
+  }
+};
```

Comment:
I guess you could just make this a one liner the same goes with the `isAfterCefBlock`.

```suggestion
const isBeforeCefBlock = (root: SugarElement, pos: CaretPosition) =>
  pos.isAtEnd() && isCefBlock(new DomTreeWalker(pos.container(), root.dom).next());
```